src/nodes.ts

```diff
@@ -283,16 +283,19 @@ export class Conditional extends Node {
 }
 
 export class Program extends Node {
+  context: ParseContext;
+
   constructor(
     line: number,
     column: number,
     start: number,
     end: number,
     raw: string,
     readonly body: Block | null,
-    readonly context: ParseContext
+    context: ParseContext
   ) {
     super('Program', line, column, start, end, raw);
+    this.context = context;
```

Comment:
Ah, right, it's the `delete programNode.context` line. (Really, it seems wrong for typescript to allow you to delete `context` in this case, but in this case I'm fine with it.)

And yep, it's nice to get rid of all of the old parser code in favor of the much nicer mapper system.